packages/next/src/build/templates/middleware.ts

```diff
@@ -18,12 +19,38 @@ if (typeof handler !== 'function') {
   )
 }
 
+// Middleware will only sent out the FetchEvent to next server,
+// so load instrumentation module here and track the error inside middleware module.
+function errorHandledHandler(fn: AdapterOptions['handler']) {
+  return async (...args: Parameters<AdapterOptions['handler']>) => {
+    try {
+      return await fn(...args)
+    } catch (error) {
+      const req = args[0]
+      edgeInstrumentationOnRequestError(
```

Comment:
```suggestion
      await edgeInstrumentationOnRequestError(
```

Reply:
Might be good to try/catch this as well and if it errors we explain the error originated from their `onRequestError` handler in instrumentation